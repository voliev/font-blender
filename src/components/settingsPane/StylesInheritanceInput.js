import React from "react";
import CheckBoxOutlineBlank from "@material-ui/icons/CheckBoxOutlineBlank";
import CheckBoxOutlined from "@material-ui/icons/CheckBoxOutlined";
import PropTypes from "prop-types";
import "./styles/StylesInheritanceInput.css";

const StylesInheritanceInput = ({
  disabled,
  element,
  inheritStyles,
  stylesInheritanceChangeHandler
}) => (
  <div className="StylesInheritanceInput">
    <input
      className="inheritance-input"
      checked={inheritStyles}
      disabled={disabled}
      name="inheritStyles"
      id={`${element}-inherit-styles`}
      onChange={stylesInheritanceChangeHandler}
      type="checkbox"
      value={element}
    />
    <label className="label" htmlFor={`${element}-inherit-styles`}>
      <div className="checkbox-icon">
        {inheritStyles ? <CheckBoxOutlined /> : <CheckBoxOutlineBlank />}
      </div>
      Inherit <code className="code"> &lt;body&gt;</code> styles
    </label>
  </div>
);

StylesInheritanceInput.propTypes = {
  disabled: PropTypes.bool,
  element: PropTypes.string.isRequired,
  inheritStyles: PropTypes.bool.isRequired,
  stylesInheritanceChangeHandler: PropTypes.func.isRequired
};

export default React.memo(StylesInheritanceInput);
