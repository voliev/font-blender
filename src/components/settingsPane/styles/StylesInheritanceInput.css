.StylesInheritanceInput {
  align-items: center;
  display: flex;
  margin-bottom: var(--space-xs);
}

.StylesInheritanceInput .label {
  align-items: center;
  display: flex;
  width: 100%;
  justify-content: flex-start;
}

.StylesInheritanceInput .checkbox-icon {
  display: flex;
  margin-right: var(--space-xs);
}

.StylesInheritanceInput .inheritance-input:disabled~.label .checkbox-icon {
  color: var(--light-grey);
}

.StylesInheritanceInput .code {
  background: var(--light-grey);
  border-radius: var(--border-radius-md);
  font-family: monospace;
  margin: 0 var(--space-xs);
  padding: 0 var(--space-xxs);
}
