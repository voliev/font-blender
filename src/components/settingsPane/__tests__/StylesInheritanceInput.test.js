import React from "react";
import { mount, shallow } from "enzyme";
import CheckBoxOutlineBlank from "@material-ui/icons/CheckBoxOutlineBlank";
import CheckBoxOutlined from "@material-ui/icons/CheckBoxOutlined";
import StylesInheritanceInput from "../StylesInheritanceInput";

const setup = (renderFn = shallow, propsOverrides = {}) => {
  const props = {
    disabled: false,
    element: "h1",
    inheritStyles: true,
    stylesInheritanceChangeHandler: jest.fn(),
    ...propsOverrides
  };

  return {
    ...props,
    wrapper: renderFn(<StylesInheritanceInput {...props} />)
  };
};

afterEach(() => {
  jest.clearAllMocks();
});

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const {
    disabled,
    element,
    inheritStyles,
    stylesInheritanceChangeHandler,
    wrapper
  } = setup(mount);

  expect(typeof wrapper.prop("disabled")).toBe("boolean");
  expect(wrapper.prop("disabled")).toBe(disabled);

  expect(typeof wrapper.prop("element")).toBe("string");
  expect(wrapper.prop("element")).toBe(element);

  expect(typeof wrapper.prop("inheritStyles")).toBe("boolean");
  expect(wrapper.prop("inheritStyles")).toBe(inheritStyles);

  expect(typeof wrapper.prop("stylesInheritanceChangeHandler")).toBe(
    "function"
  );
  expect(wrapper.prop("stylesInheritanceChangeHandler")).toBe(
    stylesInheritanceChangeHandler
  );
});

it("should set corresponding icon", () => {
  const { wrapper } = setup();

  expect(wrapper.find(CheckBoxOutlined).length).toBe(1);
  expect(wrapper.find(CheckBoxOutlineBlank).length).toBe(0);

  wrapper.setProps({ inheritStyles: false });

  expect(wrapper.find(CheckBoxOutlined).length).toBe(0);
  expect(wrapper.find(CheckBoxOutlineBlank).length).toBe(1);
});
