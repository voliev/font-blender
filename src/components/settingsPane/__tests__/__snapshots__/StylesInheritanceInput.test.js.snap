// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="StylesInheritanceInput"
>
  <input
    checked={true}
    className="inheritance-input"
    disabled={false}
    id="h1-inherit-styles"
    name="inheritStyles"
    onChange={[MockFunction]}
    type="checkbox"
    value="h1"
  />
  <label
    className="label"
    htmlFor="h1-inherit-styles"
  >
    <div
      className="checkbox-icon"
    >
      <CheckBoxOutlinedIcon />
    </div>
    Inherit 
    <code
      className="code"
    >
       &lt;body&gt;
    </code>
     styles
  </label>
</div>
`;
